specification/storage/resource-manager/Microsoft.Storage/preview/2020-08-01-preview/storage.json

```diff
@@ -0,0 +1,3793 @@
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "enum": [
+              "geoReplicationStats",
+              "blobRestoreStatus"
+            ],
+            "x-ms-enum": {
+              "name": "StorageAccountExpand",
+              "modelAsString": false
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- properties retrieved successfully for the storage account.",
+            "schema": {
+              "$ref": "#/definitions/StorageAccount"
+            }
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "StorageAccounts"
+        ],
+        "operationId": "StorageAccounts_Update",
+        "description": "The update operation can be used to update the SKU, encryption, access tier, or tags for a storage account. It can also be used to map the account to a custom domain. Only one custom domain is supported per storage account; the replacement/change of custom domain is not supported. In order to replace an old custom domain, the old value must be cleared/unregistered before a new value can be set. The update of multiple properties is supported. This call does not change the storage keys for the account. If you want to change the storage account keys, use the regenerate keys operation. The location and name of the storage account cannot be changed after creation.",
+        "x-ms-examples": {
+          "StorageAccountUpdate": {
+            "$ref": "./examples/StorageAccountUpdate.json"
+          },
+          "StorageAccountEnableAD": {
+            "$ref": "./examples/StorageAccountEnableAD.json"
+          },
+          "StorageAccountEnableCMK": {
+            "$ref": "./examples/StorageAccountEnableCMK.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/StorageAccountName"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/StorageAccountUpdateParameters"
+            },
+            "description": "The parameters to provide for the updated account."
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- storage account properties updated successfully.",
+            "schema": {
+              "$ref": "#/definitions/StorageAccount"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Storage/deletedAccounts": {
```

Comment:
Please follow keyvaults pattern.  List at subscription via fanout but then have individual GETs via a locationbased API as well.  Reference:  specification\keyvault\resource-manager\Microsoft.KeyVault\preview\2020-04-01-preview\keyvault.json and https://armmanifest.visualstudio.com/One/_git/Manifest?path=%2FMICROSOFT.KEYVAULT%2FPROD%2FMANIFEST.JSON

Reply:
We have no use for individual GETs in our use case. By design, Portal will list deleted accounts in all regions as customer is only required to provided minimal information, i.e the account name.

Is this a blocker Chris?

Reply:
Yes, it is not consistent with the other soft-delete APIs that exist today and users should be able to perform a GET on a resource ID (i.e. the resource id of the Microsoft.Storage/deletedAccounts instance) and get that resource.  The portal is only one client of your API.  It will be available via other clients and should follow standard ARM API patterns